tests/nnet/losses/test_focal_loss.py

```diff
@@ -1,13 +1,47 @@
-import numpy as np
-from numpy.testing import assert_allclose
-from hypothesis import given, assume
-import hypothesis.strategies as st
 import hypothesis.extra.numpy as hnp
+import hypothesis.strategies as st
+import numpy as np
 import pytest
+from hypothesis import given
 
-from mygrad import Tensor, sum, log
+import mygrad as mg
 from mygrad.nnet.activations import softmax
-from mygrad.nnet.losses import softmax_focal_loss, focal_loss
+from mygrad.nnet.losses import focal_loss, softmax_focal_loss
+from tests.wrappers.uber import backprop_test_factory, fwdprop_test_factory
+
+
+def numpy_focal_loss(
+    scores: np.ndarray, targets: np.ndarray, alpha: float, gamma: float
+) -> np.ndarray:
+    if isinstance(targets, mg.Tensor):
+        targets = targets.data
+    rows = np.arange((len(scores)))
+    pc = scores[rows, targets]
+    return -alpha * np.clip(1 - pc, a_min=0, a_max=1) ** gamma * np.log(pc)
+
+
+def numpy_softmax_focal_loss(
+    scores: np.ndarray, targets: np.ndarray, alpha: float, gamma: float
+) -> np.ndarray:
+    if isinstance(targets, mg.Tensor):
+        targets = targets.data
+    scores = softmax(scores).data
+    rows = np.arange((len(scores)))
```

Comment:
unnecessary extra parens?
```suggestion
    rows = np.arange(len(scores))
```